dask/dataframe/io/tests/test_parquet.py

```diff
@@ -3512,3 +3512,104 @@ def test_custom_metadata(tmpdir, engine):
             custom_metadata=custom_metadata,
         )
     assert "User-defined key/value" in str(e.value)
+
+
+def test_custom_filename(tmpdir, engine):
+    fn = str(tmpdir)
+    pdf = pd.DataFrame(
+        {"num1": [1, 2, 3, 4], "num2": [7, 8, 9, 10]},
+    )
+    df = dd.from_pandas(pdf, npartitions=2)
+    df.to_parquet(fn, datafile_name_template="hi-*.parquet", engine=engine)
+
+    files = os.listdir(fn)
+    assert "_common_metadata" in files
+    assert "_metadata" in files
+    assert "hi-0.parquet" in files
+    assert "hi-1.parquet" in files
+    assert_eq(df, dd.read_parquet(fn, engine=engine))
+
+
+def test_custom_filename_throws_error_when_append_is_true(tmpdir, engine):
```

Comment:
@rjzamora - thank you for reviewing & commenting.  I did more digging and found that a custom file name template works with `append=True` if `ignore_divisions=True` is set and the pyarrow engine is used.  It does not work when the fastparquet engine is used.  I'm not sure the exact underlying issue but will try to study up on fastparquet more to have a better idea.  Thanks again.

Reply:
>works with append=True if ignore_divisions=True is set

Cool - That makes sense.  I am interested to learn what is going wrong with the fastparquet engine, and  I appreciate your time on this!

Reply:
A quick check: for the fastparquet case, the new data gets written into hi-0.parquet instead of hi-2, but the _metadata is correctly updated to three row-groups. The first of these partitions refers to a file that has been overwritten, and the bytes are in the wrong place.

Reply:
Looks like fastparquet assumes the `part.%d.parquet` filenaming convention, [see here](https://github.com/dask/fastparquet/blob/efd3fd19a9f0dcf91045c31ff4dbb7cc3ec504f2/fastparquet/writer.py#L988).  I'm excited to look into that lib in any case, so this is a good excuse 🤓